melanoma_phd/database/variable/VariableStaticMixin.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+
+import pandas as pd
+
+from melanoma_phd.database.variable.BaseVariableConfig import BaseVariableConfig
+from melanoma_phd.database.variable.Variable import Variable
+
+
+class VariableStaticMixin:
+    """Mixin class for all static variables."""
+
+    def __init__(self, config: BaseVariableConfig) -> None:
+        super().__init__(config=config)
```

Comment:
> If this is a class without parent, it does not make sense to me that it calls to super().
Well, this is exactly the trick with `Mixin`. Using the upper class (but not inheriting from it).

On the other hand, I see more clean and less error-prone the fact that from a child class you do not need to initialize every parent class, like super(VariableStaticMixin, self).__init__() and super(CategoricalVariable, self).__init__(), in case in the mixin we have a common code in `__init__()` method, like the case of `VariableDynamicMixin`. I prefer to just call from child classes one method, super().__init__() and then the Mixin class will call the init method of the top base class, like:
CategoricalVariableStatic.__init__() -> VariableStaticMixin.__init__() -> CategoricalVariable.__init()

And not having to do:
CategoricalVariableStatic.__init__() -> VariableStaticMixin.__init__()
CategoricalVariableStatic.__init__() -> CategoricalVariable.__init__()

What do you think?

Reply:
I am unsure about it, because then you are conditioning the `Mixin` to all the combinations that the child class are going to use which I don't think I like it.

Other options could be:
1. the `__init__` of the `Mixin` to have `*args, **kwargs` and don't do anything, then the `super().__init__` won't give any error.
2. As using `super(VariableStaticMixin, self).__init__(config=config)` what it would do is do the init of all the parent classes except the `VariableStaticMixin`, then we could use this instead of individual initialization of each parent.

However, I don't think there is much pain on any of the solutions, only it was something to comment that as I though this as a red flag for the dependency of classes.

Reply:
I agree! What we have right now defined with mixins is so coupled to inheriting mixin in conjunction with concrete classes. I see it could be bad or could be an advantage for having a clear idea how they have to be used. However, I think this mixin architecture needs to be improved but so far I'd keep it like this.